pkgs/_analyzer_macros/test/golden_test.dart

```diff
@@ -69,20 +102,32 @@ void main() {
         final augmentationUnit =
             resolvedLibrary.units.singleWhere((u) => u.isMacroAugmentation);
 
-        // Each `QueryClass` outputs its query result as a comment in an
-        // augmentation. Collect them and merge them to compare with the
-        // golden.
-        final macroOutputs = augmentationUnit.content
-            .split('\n')
-            .where((l) => l.startsWith('// '))
-            .map((l) =>
-                json.decode(l.substring('// '.length)) as Map<String, Object?>);
-        macroOutput = _merge(macroOutputs);
-
-        expect(macroOutput, golden,
-            reason: updateGoldens
-                ? null
-                : '\n--> To update goldens, run: UPDATE_GOLDENS=yes dart test');
+        if (introspectionGolden != null) {
+          // Each `QueryClass` outputs its query result as a comment in an
+          // augmentation. Collect them and merge them to compare with the
+          // golden.
+          final macroOutputs = augmentationUnit.content
+              .split('\n')
+              .where((l) => l.startsWith('// '))
+              .map((l) => json.decode(l.substring('// '.length))
+                  as Map<String, Object?>);
+          introspectionMacroOutput = _merge(macroOutputs);
+
+          expect(introspectionMacroOutput, introspectionGolden,
+              reason: updateGoldens
+                  ? null
```

Comment:
Consider maybe a message like `--> Goldens will be updated, run again and test should pass`